llvm/test/Transforms/LICM/hoist-speculatable-load.ll

```diff
@@ -11,12 +11,12 @@ define void @f(i32 %ptr_i, ptr %ptr2, i1 %cond) {
 ; CHECK-NEXT:    store i32 0, ptr [[PTR2:%.*]], align 4
 ; CHECK-NEXT:    br label [[FOR_BODY_LR_PH]]
 ; CHECK:       for.body.lr.ph:
-; CHECK-NEXT:    [[TMP0:%.*]] = load i32, ptr [[PTR]], align 4
```

Comment:
I think you'd preserve test intent by adding nofree nosyc to this one (maybe with a name change), and then adding a separate one without it.

It isn't clear to my why with the additional attributes we still can't hoist the load.  That's concerning?

Reply:
I added `f_nofree_nosync` below; the problem seems to be that `%ptr` is formed using `inttoptr` and `Value::canBeFreed` only applies to function arguments I think (it would be valid for the function to allocate and free locally).

So eventually we will probably need more powerful analysis.

Reply:
Maybe add a variant which doesn't involve the inttoptr then?